src/style-spec/reference/v8.json

```diff
@@ -2761,7 +2761,7 @@
         }
       },
       "coalesce": {
-        "doc": "Evaluates each expression in turn until the first non-null value is obtained, and returns that value.",
+        "doc": "Evaluates each expression in turn until the first valid value is obtained. Invalid values are `null` and [`'image'`](#types-image) expressions that are unavailable in the style. If all values are invalid, coalesce returns the first value listed.",
```

Comment:
>If all values are invalid, coalesce returns the first value listed.

I think it might be more useful and accurate to say something like `If all values are invalid, coalesce returns null, except when image expressions are used, in which case the styleimagemissing event will be fired for the first image in the list.`

Reply:
I see where you're coming from, but there's two edge-cases that I'm not sure that covers: `coalesce` expressions that mix `image` and `null` values (not sure if this is a real use-case) and expressions where the expression is used to style another property based on whether the image is present or missing. (We have at least two issues concerning this use case.)

The `coalesce` expression is currently the only way to allow an image's absence to change other paint and layout properties. If I'm not mistaken, `styleimagemissing` is only fired if an unavailable image is passed into an image property, so in these cases it would not be fired.